### ISSUE TITLE
[Bug] A couple of bugs in Arcane Dimensions

### ISSUE THREAD

**Issue by TMHKR** [Novum/vkQuake#426]
**Bug 1:** Some sounds specific to AD won't play (examples: breakable windows, flying skulls and necromancers in ad_necrokeep)
**Bug 2:** Difficulty level shown on HUD when pressing the Tab key remains "skill 1" regardless of what difficulty you chose

vkQuake: 1.12.2
Arcane Dimensions: 1.80 patch 1
OS: Windows 10 x64

**Comment by Novum** [comment]
One bug per issue please.

**Comment by TMHKR** [comment]
I apologize (new on this website) but at least I hope I made you aware of these bugs.

**Comment by vsonnier** [comment]
> Bug 1: Some sounds specific to AD won't play (examples: breakable windows, flying skulls and necromancers in ad_necrokeep)

Works for me. Load save, break the window, the flying skulls are making noise. Both for 1.12.2 and the current `master`.
[s0.zip](https://github.com/Novum/vkQuake/files/8043649/s0.zip)

**Comment by Novum** [comment]
@vsonnier I feel like I had this happen before to me that certain sounds aren't playing. So I'll keep this open for now, need to investigate at some point.

**Comment by TMHKR** [comment]
**UPDATE:** Bug 1 (sound bug) doesn't seem to occur by using cl_nopext 0 (I had it set to 1)

**Comment by vsonnier** [comment]
> UPDATE: Bug 1 (sound bug) doesn't seem to occur by using cl_nopext 0 (I had it set to 1)

I see that on the `master` too with `cl_nopext 0`. Spiked seems to have the same problem. On the other hand, Ironwail (which has no extensions support at all) works as expected.

**Comment by vsonnier** [comment]
OK, I think the bug is here:

https://github.com/Novum/vkQuake/blob/c8721b2f3bc38d4027b895e2f568873d8a6ef370/Quake/sv_main.c#L1216

This is a much simpler condition that in regular Quakspasm when testing sound capabilities vs. protocols.

Here, as in Spiked, if we have `cl_nopext 1`, then `client->protocol_pext2` is 0 and if we have more than 256 cached sounds (which we have here) this teh sound play is bailed out like if we were in protocol 15.

**Comment by vsonnier** [comment]
Question : 

In `sv_main.c (1216)`, if we change the check by simply:
```c
//PROTOCOL_NETQUAKE do not support more than 256 sounds and/or 8192 entities.
if ((field_mask & (SND_LARGEENTITY | SND_LARGESOUND)) && (sv.protocol == PROTOCOL_NETQUAKE))
      continue;
``` 

This will effectively disable all sound as (256+ num, 8192+ entity) like in regular Quakespasm for protocol 15 (PROTOCOL_NETQUAKE). On the other hand, the other remaining protocols 666 / 999 are capable of playing either large sounds / large entity, whatever extensions are on or off (as in Quakespasm), so no problem.

Now if I understand correctly reading the code an coments, we could also have extensions on protocol 15, but I don't see in such hybrid situation what should be authorized : large entities ? large sounds ? more than 8 channels ?

**Comment by vsonnier** [comment]
I've made a PR #435 that fix the problem while offering the same limitations as the origival Quakespasm.

**Comment by Novum** [comment]
I cannot reproduce the skill bug, closing this now.

**Comment by vsonnier** [comment]
> I cannot reproduce the skill bug, closing this now.

The skill bug was solved by #428
